tests/run_examples.py

```diff
@@ -313,7 +287,7 @@ def run(exdir, logdir, args):
 
 
 def run_all(args):
-    if args.assert_gpu and "gpu" not in available_devices:
+    if args.assert_gpu and dpctl_available and "gpu" not in available_devices:
```

Comment:
Generally, this is wrong, because we have an option to offload GPU without dpctl.
See `config_context(target_offload=True)`

Reply:
Check of the GPU availability depends on GPU, there should be another mechanism for that I think.

Reply:
At least here check for `dpctl_available` is unnecessary.

Reply:
This was originally added because there is concern when dpctl is present but gpu not in available devices, since dpctl is only installed for GPU testing

Reply:
I think, here check for `dpctl_available` is unnecessary. In the top of the file it was done for `available_devices`.
We can leave it as is here.
```suggestion
    if args.assert_gpu and "gpu" not in available_devices:
```

Reply:
Its because of assert_gpu logic. But it makes sense to do this and revise infra logic, done.